libraries/oneview_helper.rb

```diff
@@ -18,31 +18,39 @@ module Helper
     # @param context Context from the resource action block (self)
     # @param [String, Symbol] type Name of the resource type. e.g., :EthernetNetwork
     # @param [String, Symbol] action Action method to call on the resource. e.g., :create_or_update
-    def self.do_resource_action(context, type, action)
-      klass = OneviewCookbook::Helper.get_resource_class(context, type)
+    # @param [Module] base_module The base API module supported. e.g. OneviewCookbook::ImageStreamer
+    def self.do_resource_action(context, type, action, base_module = OneviewCookbook)
+      klass = OneviewCookbook::Helper.get_resource_class(context, type, base_module)
       res = klass.new(context)
       res.send(action)
     end
 
     # Get the resource class by name, taking into consideration the resource's properties
     # @param context Context from the resource action block (self)
     # @param [String, Symbol] type Name of the resource type. e.g., :EthernetNetwork
+    # @param [Module] base_module The base API module supported. e.g. OneviewCookbook::ImageStreamer
     # @return [Class] Resource class
-    def self.get_resource_class(context, resource_type)
+    def self.get_resource_class(context, resource_type, base_module = OneviewCookbook)
       load_sdk(context)
-      api_version = context.property_is_set?(:api_version) ? context.api_version : context.node['oneview']['api_version']
-      api_module = get_api_module(api_version)
+      # Loads the api version if the property was specified directly
+      context_api_version = context.api_version if context.property_is_set?(:api_version)
+      # Loads the api version if it was specified in the client
+      client_api_version = convert_keys(context.client, :to_sym)[:api_version] if context.property_is_set?(:client)
+      # Loads the api version giving preference: property > client > node default
+      api_version = context_api_version || client_api_version || context.node['oneview']['api_version']
```

Comment:
Not exactly ignored. It will get the client version before the client is built.

So the client will only have api_version by default if it is a `OneviewSDK::Client`, and if the user is creating directly the `OneviewSDK::Client` he must know the implications.

I agree it would be a breaking change, but I guess it won't be a problem.